modules/dnn/CMakeLists.txt

```diff
@@ -165,24 +165,32 @@ elseif(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
 endif()
 
 set(dnn_runtime_libs "")
-if(INF_ENGINE_TARGET)
-  set(use_nn_builder OFF)
-  if(TARGET inference_engine_nn_builder OR # custom imported target
-     TARGET IE::inference_engine_nn_builder OR # default imported target via InferenceEngineConfig.cmake
-     INF_ENGINE_RELEASE VERSION_LESS "2020000000") # compatibility with older versions on IE
-    set(use_nn_builder ON)
+if(TARGET openvino::runtime)
+  message(STATUS "DNN: Enabling OpenVINO Toolkit support")
+  math(EXPR OV_VER "${OpenVINO_VERSION_MAJOR} * 1000000 + ${OpenVINO_VERSION_MINOR} * 10000 + ${OpenVINO_VERSION_PATCH} * 100")
+  # TODO: rename definitions
+  add_definitions("-DHAVE_DNN_NGRAPH" "-DHAVE_INF_ENGINE=1" "-DINF_ENGINE_RELEASE=${OV_VER}")
```

Comment:
G-API also uses `INF_ENGINE_RELEASE` variable.

Could we reuse `ocv_add_external_target()` approach with added definitions?

Reply:
Done! I've added `ocv.3rdparty.openvino` interface target with dependencies on IE and ngraph or whole OpenVINO. It seem to be working with OpenVINO@_master_ and [2021.4 package](https://storage.openvinotoolkit.org/repositories/openvino/packages/2021.4/).

Also removed `OPENCV_DNN_IE_NN_BUILDER_2019` option from cmake. I did not remove it from the code to avoid conflicts.

Apparently, `-DWITH_INF_ENGINE=ON -DWITH_NGRAPH=OFF` does not work on current _4.x_ branch (with 2021.4), I didn't try to fix it. Looks like nobody uses such configuration and future OpenVINO versions will not have separation between these components.